tests/Feature/GroupPermissionsTest.php

```diff
@@ -130,4 +130,54 @@ public function testUserCannotSeePrivateGroups()
         $response->assertStatus(200);
         $response->assertJsonMissing(['id' => $group->id]);
     }
+
+    public function testLeaderCanPromoteMember()
+    {
+        $leader = factory(User::class)->create([
+            'role_id' => Role::where('name', 'User')->first()->id,
+        ]);
+
+        $user = factory(User::class)->create([
+            'role_id' => Role::where('name', 'User')->first()->id,
+        ]);
+
+        $group = factory(Group::class)->create([
+            'user_id' => $leader->id
+        ]);
+
+        $this->actingAs($leader);
+
+        $response = $this->put("web/group/$group->id/promote", [
+            'user_id' => $user->id
+        ]);
+
+        $response->assertStatus(200);
```

Comment:
We need to also verify that the changes did take effect.
```suggestion
        $response->assertStatus(200);
        
        // Verify that the leader has been changed
        $group->fresh();
        $this->assertEqual($user->id, $group->user_id);
```